src/edge/simulator.ts

```diff
@@ -91,85 +52,94 @@ export class Simulator {
-                } catch (error) {
-                    if (outputChannel) {
-                        outputChannel.appendLine(`${Constants.failedInstallSimulator} ${error.message}`);
-                    }
-                    ret =  InstallReturn.Failed;
-                }
-            } else {
-                ret = InstallReturn.Canceled;
-            }
-        } else {
-            ret = InstallReturn.NoPip;
-        }
-        return ret;
+    private static async getLastestStandaloneSimulatorInfo() {
+        const infoString = await request.get(Simulator.latestReleaseInfoUrl, {
+            headers: {
+                "User-Agent": "vscode-azure-iot-edge",
+            },
+        });
+        const infoJson = JSON.parse(infoString);
+        return infoJson;
     }
 
-    private static async simulatorInstalled(): Promise<boolean> {
-        return await Simulator.checkCmdExist("iotedgehubdev");
+    private static adjustTerminalCommand(command: string): string {
+        return (os.platform() === "linux" || os.platform() === "darwin") ? `sudo ${command}` : command;
     }
 
-    private static async checkPipInstalled(): Promise<boolean> {
-        return await Simulator.checkCmdExist("pip");
+    private isInstalling: boolean = false;
+
+    constructor(private context: vscode.ExtensionContext) {
     }
 
-    private static async checkCmdExist(cmd: string): Promise<boolean> {
+    public async validateSimulatorUpdated(outputChannel: vscode.OutputChannel = null): Promise<void> {
+        let message: string;
+        let type: string = "";
+        const telemetryName = "simulatorUpdated";
         try {
-            await Executor.executeCMD(null, cmd, { shell: true }, "--version");
-            return true;
+            const simulatorType = await this.getSimulatorType();
+            if (simulatorType === SimulatorType.NotInstalled) {
+                message =  Constants.needSimulatorInstalledMsg;
+                type = "install";
+            } else {
+                const version: string | null = await this.getCurrentSimulatorVersion();
+                if (version && semver.valid(version)) {
+                    const latestVersion: string | undefined = await this.getLatestSimulatorVersion();
+                    if (latestVersion && !semver.eq(latestVersion, version)) {
+                        message = `${Constants.updateSimulatorMsg} (${version} to ${latestVersion})`;
+                    }
+                } else {
+                    message = Constants.updateSimulatorMsg;
+                }
+
+                if (simulatorType === SimulatorType.Pip) {
+                    type = "upgradePipPackage";
+                } else {
+                    type = "upgradeStandalone";
+                }
+            }
         } catch (error) {
-            return false;
+            message =  Constants.needSimulatorInstalledMsg;
```

Comment:
when will error be thrown?

Reply:
Checked and found out that we do not need to handle error